packages/admin/admin-date-time/src/DateRangePicker.tsx

```diff
@@ -0,0 +1,91 @@
+import "react-date-range/dist/styles.css";
+import "react-date-range/dist/theme/default.css";
+
+import { InputWithPopper, InputWithPopperProps } from "@comet/admin";
+import * as React from "react";
+import { DateRange as ReactDateRange, DateRangeProps as ReactDateRangeProps, Range } from "react-date-range";
+import { FormatDateOptions, useIntl } from "react-intl";
+
+type DateRangePickerComponentsProps = InputWithPopperProps["componentsProps"] & {
+    dateRange?: Partial<Omit<ReactDateRangeProps, "onChange" | "ranges">>;
+};
+
+export type DateRange = {
+    start: Date;
+    end: Date;
+};
+
+export interface DateRangePickerProps extends Omit<InputWithPopperProps, "children" | "value" | "onChange" | "componentsProps"> {
+    onChange?: (range: DateRange) => void;
+    value?: DateRange;
+    formatDateOptions?: FormatDateOptions;
+    rangeStringSeparator?: string;
+    componentsProps?: DateRangePickerComponentsProps;
+}
+
+const useDateRangeTextValue = (range: DateRange | null | undefined, rangeStringSeparator: string, formatDateOptions?: FormatDateOptions): string => {
+    const intl = useIntl();
+
+    if (range) {
+        const startDateString = intl.formatDate(range.start, formatDateOptions);
+        const endDateString = intl.formatDate(range.end, formatDateOptions);
+
+        if (startDateString === endDateString) {
+            return startDateString;
+        }
+
+        return `${startDateString}${rangeStringSeparator}${endDateString}`;
+    }
+
+    return "";
+};
+
+const rangeKey = "pickedDateRange";
+
+const getRangeFromValue = (value: undefined | DateRange): Range => {
+    if (value?.start) {
+        return {
+            key: rangeKey,
+            startDate: value.start,
+            endDate: value.end ?? value.start,
+        };
+    }
+
+    return {
+        key: rangeKey,
+        startDate: new Date(),
+        endDate: new Date(),
```

Comment:
This happens when no `value` is defined. `react-date-range` always requires a value to display, otherwise, it highlights everything as selected. To prevent this, I just set the current date as the range, this only applies to the visuals. The `value` is not changed until the user manually selects a range.

Reply:
ok.... do you think it might be worth explaining that in a comment?

Reply:
Yes, that probably makes sense 👍🏼